### PR TITLE
adds api error handling example to documentation

### PR DESCRIPTION
**Pull request by LBWright**
Describe what this patch does to fix the issue.

This adds documentation to the error handling page. I was having a hard time finding anything on the proper use of abort with the flask error handler. After settling on a good pattern, I added it here.

Link to any relevant issues or pull requests.

### CODE REVIEW

File: docs/patterns/errorpages.rst [hunk 1]
```diff
@@ -97,3 +97,30 @@ An example template might be this:
...
+When using Flask for web APIs, handling errors is as simple as the previous examples but with minor changes.
+
+Register the error handler::
+
+    from flask import jsonify
+
+    @app.errorhandler(404)
+    def resource_not_found(e):
+        # if passing in an exception directly, you may need to convert it to a string
+        return jsonify(error=e)
```
Comment (davidism) [inline docs/patterns/errorpages.rst]: As far as I know, `json.dumps` won't be able to serialize an `Exception` object, so this won't work.

Comment (LBWright) [reply docs/patterns/errorpages.rst]: You're right. That was the comment in the example. I'll update the code to be a little more clear.

Comment (LBWright) [reply docs/patterns/errorpages.rst]: Added more clarity.

File: docs/patterns/errorpages.rst [hunk 2]
```diff
@@ -97,3 +97,31 @@ An example template might be this:
...
+
+Register the error handler::
+
+    from flask import jsonify
+
+    @app.errorhandler(404)
+    def resource_not_found(e):
+        # if passing in an Exception object directly, you may need to convert it to a string
+        stringified_exception = str(e)
+        return jsonify(error=stringified_exception)
```
Comment (davidism) [inline docs/patterns/errorpages.rst]: This no longer returns a 404 status.

Comment (LBWright) [reply docs/patterns/errorpages.rst]: Thanks. I'll rebase to squash these commits down and update.

**Pull request merged**